orchagent/mirrororch.cpp

```diff
@@ -562,175 +601,165 @@ bool MirrorOrch::updateSessionDstPort(const string& name, MirrorEntry& session)
-            continue;
-        }
-
-        if (session.status)
-        {
-            if (!updateSessionDstMac(name, session))
-            {
-                continue;
-            }
-
-            if (!updateSessionDstPort(name, session))
-            {
-                continue;
-            }
+            session.nexthopInfo.nexthop = *update.nexthopGroup.getIpAddresses().begin();
         }
         else
         {
-            if (!activateSession(name, session))
-            {
-                continue;
-            }
+            session.nexthopInfo.nexthop = IpAddress();
         }
+
+        updateSession(name, session);
     }
 }
 
 void MirrorOrch::updateNeighbor(const NeighborUpdate& update)
 {
     SWSS_LOG_ENTER();
 
-    auto sessionIter = m_syncdMirrors.begin();
-    for (; sessionIter != m_syncdMirrors.end(); ++sessionIter)
+    for (auto it = m_syncdMirrors.begin(); it != m_syncdMirrors.end(); it++)
     {
-        if (!sessionIter->second.nexthopInfo.resolved)
-        {
-            continue;
-        }
-
-        // It is possible to have few sessions that points to one next hop
-        if (sessionIter->second.nexthopInfo.nexthop != update.entry.ip_address)
-        {
-            continue;
-        }
+        const auto& name = it->first;
+        auto& session = it->second;
 
-        SWSS_LOG_INFO("Updating neighbor info %s %s\n", update.entry.ip_address.to_string().c_str(), update.entry.alias.c_str());
+        SWSS_LOG_NOTICE("neighbor %s : %s", update.entry.ip_address.to_string().c_str(), update.entry.alias.c_str());
 
-        const auto& name = sessionIter->first;
-        auto& session = sessionIter->second;
 
-        if (update.add)
+        // XXX: add comments!
+        if (session.dstIp != update.entry.ip_address &&
+                session.nexthopInfo.nexthop != update.entry.ip_address)
         {
-            if (!getNeighborInfo(name, session, update.entry, update.mac))
-            {
-                if (session.status)
-                {
-                    deactivateSession(name, session);
-                }
-               continue;
-            }
+            continue;
+        }
 
-            if (session.status)
-            {
-                if (!updateSessionDstMac(name, session))
-                {
-                    continue;
-                }
+        SWSS_LOG_NOTICE("Updating mirror session %s with neighbor %s",
+                name.c_str(), update.entry.alias.c_str());
 
-                if (!updateSessionDstPort(name, session))
-                {
-                    continue;
-                }
-            }
-            else
-            {
-                activateSession(name, session);
-            }
-        }
-        else if (session.status)
-        {
-            deactivateSession(name, session);
-            session.neighborInfo.resolved = false;
-        }
+        updateSession(name, session);
     }
 }
 
 void MirrorOrch::updateFdb(const FdbUpdate& update)
```

Comment:
>updateFdb [](start = 17, length = 9)

can you add a note to this function. what does it try to achieve?

when we receive a fdb notification:
1 mac move: then we could protentially update the dst port of the session
2. mac remove: make the session inactive.